app/controllers/spree/admin/orders_controller.rb

```diff
@@ -37,6 +37,11 @@ def edit
       def update
         @order.recreate_all_fees!
 
+        unless @order.cart?
+          @order.create_tax_charge!
+          @order.update!
+        end
```

Comment:
it may be too much to ask to extract a service to avoid adding yet more responsibilities to the controller, but this is growing and we should do it soon IMO.

Reply:
:100: Yeah I think ultimately we need a new service (or two) that handles this stuff with updating orders after they've changed. There's a lot of logic around that which is currently spread out. There's a couple more changes related to the new `Spree::ItemAdjustments` class I'd like to bring in next (and then the adjustments changes will basically be finished) and then it would be really good to assess how we can improve these things.

Reply:
but my point is, defining some boundaries by improving the design should make this easier to change. I know it would then not be possible to apply commits Spree's into our repo but it'd help us in the long run.